From 11093e427219a3d1a855ce33b64848fb2c4c23f0 Mon Sep 17 00:00:00 2001
From: Julian Greilich <j.greilich@gmx.de>
Date: Mon, 18 May 2026 15:02:49 +0200
Subject: a11y Windows: Notify system about changes of the expanded state

Expanding/Collapsing an expandable control now leads to the correct
announcement of the current expanded state.

Task-number: QTBUG-146552
Change-Id: I78d8b09e0db01cedc41fdb1d1b60c3feafd11016
---
 .../windows/uiautomation/qwindowsuiamainprovider.cpp | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git x/qtbase/src/plugins/platforms/windows/uiautomation/qwindowsuiamainprovider.cpp y/qtbase/src/plugins/platforms/windows/uiautomation/qwindowsuiamainprovider.cpp
index 675825c4d785113ec2174a17155f98d0d99e7bc7..92c3997922c5d9407e74af22382d4ceced187143 100644
--- x/qtbase/src/plugins/platforms/windows/uiautomation/qwindowsuiamainprovider.cpp
+++ y/qtbase/src/plugins/platforms/windows/uiautomation/qwindowsuiamainprovider.cpp
@@ -98,6 +98,18 @@ void QWindowsUiaMainProvider::notifyStateChange(QAccessibleStateChangeEvent *eve
                 }
             }
         }
+        if (event->changedStates().expanded) {
+            if (accessible->state().expandable) {
+                if (auto provider = providerForAccessible(accessible)) {
+                    long expandedState = accessible->state().expanded ? ExpandCollapseState_Expanded : ExpandCollapseState_Collapsed;
+
+                    QComVariant oldVal;
+                    QComVariant newVal{expandedState};
+                    UiaRaiseAutomationPropertyChangedEvent(
+                            provider.Get(), UIA_ExpandCollapseExpandCollapseStatePropertyId, oldVal.get(), newVal.get());
+                }
+            }
+        }
         if (event->changedStates().active) {
             if (accessible->role() == QAccessible::Window) {
                 // Notifies window opened/closed.
